### PR TITLE
 #403: transformed e2e scores tests from cypress to pytest

### PR DESCRIPTION
**Pull request by cyplas**
### Detailed information:

Rewrote the scores e2e tests from cypress to python.
Refactored uuid validation to increase reusability.

### Closing issues: 

List all issues the pull request solve: 

- *#403*

### Test plan (required)

- [X] The PR contains new PyTest unit/integration tests for any function or functional added. 
- [X] The PR changes existing PyTest unit/integration tests to keep all tests up to date.
- [X] The PR does not lead to degradation in unit test coverage.
- [X] Project parts affected by changes in this PR was tested manually on your local (using Postman or in any other way). List everything you've tested below:
  - pytest

### CODE REVIEW

File: app/scoring/tests/test_scores_routes.py
Comment (cyplas) [inline app/scoring/tests/test_scores_routes.py:8]: Regarding this TODO, I haven't moved the files yet, as they are used by other cypress tests yet to be rewritten.

File: app/scoring/tests/test_scores_routes.py
Comment (danmash) [inline app/scoring/tests/test_scores_routes.py:12]: I'm sure we have some function to check if uuid is valid or no, no need to create a new one

**Review by danmash — CHANGES_REQUESTED** [review]
I appreciate you've checked examples in the repo and used some stuff from it like "client_with_user_and_header". Nicely done 👍

**Comment by danmash** [comment]
Hi @cyplas 
will you be able to adjust this?

File: app/scoring/tests/test_scores_routes.py
Comment (cyplas) [reply app/scoring/tests/test_scores_routes.py:12]: > I'm sure we have some function to check if uuid is valid or no, no need to create a new one

We do have app.common.validate_uuid, but it raises one of our CustomErrors, which have HTTP status_codes, which are not relevant here. But perhaps that is too conservative and I should just go ahead and use it? In principle I agree with you.

File: app/scoring/tests/test_scores_routes.py
Comment (cyplas) [reply app/scoring/tests/test_scores_routes.py]: Good idea, I've copied them to under common/, as they are also used by other e2e tests.

File: app/scoring/tests/test_scores_routes.py
Comment (danmash) [reply app/scoring/tests/test_scores_routes.py:10]: So basically you have the same logic `try-except + uuid.UUID(string)`. It's a code duplication.
I recommend you move your new function to the common folder and use it here and in `app.common.validate_uuid`.

Comment (cyplas) [reply app/scoring/tests/test_scores_routes.py]: Agreed. Finally addressed this in today's commits.

I did notice something else, however, but it seems out of scope here and I suggest creating a separate issue: app.common.uuid.validate_uuid (which is used in many places) takes two arguments, but its checks (only) apply to the first one and it's error messages (only) refer to the second one.

Comment (cyplas) [reply app/scoring/tests/test_scores_routes.py]: > I did notice something else, however, but it seems out of scope here and I suggest creating a separate issue: app.common.uuid.validate_uuid (which is used in many places) takes two arguments, but its checks (only) apply to the first one and it's error messages (only) refer to the second one.

Though I don't know, maybe this was intentional, as the second argument does make the error messages a little more precise ...

**Review by danmash — APPROVED** [review]
Awesome @cyplas! 
The single thing I would like to change is moving .json files to `app/scoring/tests/data` since you are using them in the single test at the moment, these files are not common. You are free to merge after fixing this without new approval.

**Pull request merged**
